test/test_groundpropagator/test_groundpropagator.cpp

```diff
@@ -0,0 +1,295 @@
+#include <stdio.h>
+#include <cstdint>
+#include <limits>
+#include <lin.hpp>
+#include <gnc/constants.hpp>
+#include <gnc/config.hpp>
+
+//UTILITY MACROS
+#include <unity.h>
+#include "../custom_assertions.hpp"
+#include <orb/Orbit.h>
+#include <orb/GroundPropagator.h>
+
+//0 or 1 that can't be calculated at compile time.
+#ifdef DESKTOP
+#define notcompiletime (rand()%2)
+#else
+#include <Arduino.h>
+#define notcompiletime (millis()%2)
+#endif
+
+//grace orbit initial
+const orb::Orbit gracestart(uint64_t(gnc::constant::init_gps_week_number)*gnc::constant::NANOSECONDS_IN_WEEK,{-6522019.833240811L, 2067829.846415895L, 776905.9724453629L},{941.0211143841228L, 85.66662333729801L, 7552.870253470936L});
+
+//grace orbit 100 seconds later
+const orb::Orbit grace100s(uint64_t(gnc::constant::init_gps_week_number)*gnc::constant::NANOSECONDS_IN_WEEK+100'000'000'000ULL,{-6388456.55330517L, 2062929.296577276L, 1525892.564091281L},{1726.923087560988L, -185.5049475128178L, 7411.544615026139L});
+
+//grace orbit 13700 seconds later
+const orb::Orbit grace13700s(uint64_t(gnc::constant::init_gps_week_number)*gnc::constant::NANOSECONDS_IN_WEEK+13700'000'000'000ULL,{1579190.147083268L, -6066459.613667888L, 2785708.976728437L},{480.8261476296949L, -3083.977113497177L, -6969.470748202005L});
+
+//earth rate in ecef (rad/s)
+lin::Vector3d earth_rate_ecef= {0.000000707063506E-4,-0.000001060595259E-4,0.729211585530000E-4};
+
+void test_basic_constructors() {
+    orb::GroundPropagator est;
+    orb::Orbit x= est.bestestimate();
+    TEST_ASSERT_FALSE(x.valid());
+    TEST_ASSERT_EQUAL_INT(est.totalnumgravcallsleft(),0);
+}
+
+
+/** Returns real fake orbits sent from ground.*/
+orb::Orbit real_fake_ground_data(int controlcycle){
+    static orb::Orbit r=gracestart;
+    switch (controlcycle)
+    {
+    case 100:
+        //return an old orbit
+        r=gracestart;
+        return r;
+        break;
+    case 110:
+        //return an old orbit
+        r.startpropagating(r.nsgpstime()+100'000'000'000ULL,earth_rate_ecef);
+        r.finishpropagating();
+        return r;
+        break;
+    case 112:
+        //return an old orbit
+        r.startpropagating(r.nsgpstime()+1'000'000'000ULL,earth_rate_ecef);
+        r.finishpropagating();
+        return r;
+        break;
+    case 113:
+        //return an old orbit
+        r.startpropagating(r.nsgpstime()+20'000'000'000ULL,earth_rate_ecef);
+        r.finishpropagating();
+        return r;
+        break;
+    case 400:
+        //return an old orbit
+        r= grace100s;
+        r.startpropagating(r.nsgpstime()+1000'000'000'000ULL,earth_rate_ecef);
+        r.finishpropagating();
+        return r;
+        break;
+    default:
+        return orb::Orbit();
+        break;
+    }
+}
+
+
+void test_normal_use() {
```

Comment:
These tests are kind of hard to read. Consider adding more spacing and some header comments on each test describing: the purpose of the tests, the test steps, and how the steps demonstrate the purpose.

Reply:
Consider breaking out common initializations into a TestFixture class, rather than copy-pasting the initializations into every test.